contracts/modules/governance/NXMaster.sol

```diff
@@ -87,17 +87,33 @@ contract NXMaster is INXMMaster {
     require(contractAddress != address(0), "NXMaster: Contract address is 0");
 
     contractCodes.push(contractCode);
-
     address payable newInternalContract;
-    if (_type == uint(ContractType.Replaceable)) {
+    uint contractType = uint8(_type);
+
+    if (contractType == uint(ContractType.Replaceable)) {
 
       newInternalContract = contractAddress;
       isReplaceable[contractCode] = true;
-    } else if (_type == uint(ContractType.Proxy)) {
 
-      newInternalContract = payable(new OwnedUpgradeabilityProxy(contractAddress));
+    } else if (contractType == uint(ContractType.Proxy)) {
```

Comment:
can we have a test that verifies this address is predictable and asserts it?

since that was the original goal

Reply:
[#721](https://github.com/NexusMutual/smart-contracts/issues/721)